src/SurfaceSchemes/surface_conditions.jl

```diff
@@ -0,0 +1,351 @@
+    thermo_params = CAP.thermodynamics_params(params)
+    surface_params = CAP.surface_fluxes_params(params)
+
+    (!isnothing(q_vap) && atmos.moisture_model isa DryModel) &&
+        error("surface q_vap cannot be specified when using a DryModel")
+
+    if isnothing(T)
+        # Assume an idealized latitude-dependent surface temperature.
+        if coordinates isa Geometry.LatLongZPoint ||
+           coordinates isa Geometry.LatLongPoint
+            T = FT(271) + FT(29) * exp(-coordinates.lat^2 / (2 * 26^2))
+        else
+            # Assume that the latitude is 0.
+            T = FT(300)
+        end
+    end
+    if isnothing(u)
+        u = FT(0)
+    end
+    if isnothing(v)
+        v = FT(0)
+    end
+    if isnothing(p)
+        # Assume an adiabatic profile with constant cv and R above the surface.
+        cv = TD.cv_m(thermo_params, interior_ts)
+        R = TD.gas_constant_air(thermo_params, interior_ts)
+        interior_ρ = TD.air_density(thermo_params, interior_ts)
+        interior_T = TD.air_temperature(thermo_params, interior_ts)
+        ρ = interior_ρ * (T / interior_T)^(cv / R)
+        if atmos.moisture_model isa DryModel
+            ts = TD.PhaseDry_ρT(thermo_params, ρ, T)
+        else
+            if isnothing(q_vap)
+                # Assume that the surface is water with saturated air directly
+                # above it.
+                phase = TD.Liquid()
+                q_vap = TD.q_vap_saturation_generic(thermo_params, T, ρ, phase)
+            end
+            q = TD.PhasePartition(q_vap)
+            ts = TD.PhaseNonEquil_ρTq(thermo_params, ρ, T, q)
+        end
+    else
+        if atmos.moisture_model isa DryModel
+            ts = TD.PhaseDry_pT(thermo_params, p, T)
+        else
+            if isnothing(q_vap)
+                # Assume that the surface is water with saturated air directly
+                # above it.
+                phase = TD.Liquid()
+                p_sat = TD.saturation_vapor_pressure(thermo_params, T, phase)
+                ϵ_v =
+                    TD.Parameters.R_d(thermo_params) /
+                    TD.Parameters.R_v(thermo_params)
+                q_vap = ϵ_v * p_sat / (p - p_sat * (1 - ϵ_v))
+            end
+            q = TD.PhasePartition(q_vap)
+            ts = TD.PhaseNonEquil_pTq(thermo_params, p, T, q)
+        end
+    end
+
+    surface_values = SF.SurfaceValues(coordinates.z, SA.SVector(u, v), ts)
+    interior_values = SF.InteriorValues(
+        interior_z,
+        SA.SVector(interior_u, interior_v),
+        interior_ts,
+    )
+
+    if parameterization isa ExchangeCoefficients
+        if isnothing(gustiness)
+            gustiness = FT(1)
+        end
+        if isnothing(beta)
+            beta = FT(1)
+        end
+        inputs = SF.Coefficients(
+            interior_values,
+            surface_values,
+            parameterization.Cd,
+            parameterization.Ch,
+            FT(NaN), # TODO: Remove z0m from SF.Coefficients
```

Comment:
Why are we putting NaNs here?

Reply:
These are just placeholders - they aren't actually used by SurfaceFluxes but are set as NaNs so it's clear that they shouldn't be used.